src/m365/pa/commands/app/app-permission-list.ts

```diff
@@ -0,0 +1,131 @@
+import { Logger } from '../../../../cli/Logger';
+import GlobalOptions from '../../../../GlobalOptions';
+import { formatting } from '../../../../utils/formatting';
+import { odata } from '../../../../utils/odata';
+import { validation } from '../../../../utils/validation';
+import PowerAppsCommand from '../../../base/PowerAppsCommand';
+import commands from '../../commands';
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  appName: string;
+  asAdmin: boolean;
+  environment?: string;
+  roleName?: string;
+}
+
+class PaAppPermissionListCommand extends PowerAppsCommand {
+  private allowedRoleNames = ['Owner', 'CanEdit', 'CanView'];
+
+  public get name(): string {
+    return commands.APP_PERMISSION_LIST;
+  }
+
+  public get description(): string {
+    return 'Lists all permissions of a Power Apps app';
+  }
+
+  public defaultProperties(): string[] | undefined {
+    return ['roleName', 'principalId', 'principalType'];
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+    this.#initOptionSets();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        asAdmin: !!args.options.asAdmin,
+        environmentName: typeof args.options.environment !== 'undefined',
```

Comment:
Wrong telemetry name here. There's a chance we have to rename this option: https://github.com/pnp/cli-microsoft365/issues/4467#issuecomment-1619213795